server/form-pages/utils/defaultMatchingInformationValues.test.ts

```diff
@@ -178,6 +190,48 @@ describe('defaultMatchingInformationValues', () => {
     })
   })
 
+  describe('isSuitedForSexOffenders', () => {
+    it('is set to the original value if defined', () => {
+      expect(
+        defaultMatchingInformationValues(
+          { ...bodyWithUndefinedValues, isSuitedForSexOffenders: 'desirable' },
+          assessment,
+        ),
+      ).toEqual(expect.objectContaining({ isSuitedForSexOffenders: 'desirable' }))
+    })
+
+    const truthyValues = [['current'], ['previous'], ['current', 'previous']]
+
+    truthyValues.forEach(value => {
+      it.each(sexualOffencesFields)(
+        `is set to 'essential' when there's no original value and \`%s\` === ['${value.join("', '")}']`,
+        testedField => {
+          sexualOffencesFields.forEach(field =>
+            when(retrieveOptionalQuestionResponseFromFormArtifact)
+              .calledWith(assessment.application, DateOfOffence, field)
+              .mockReturnValue(testedField === field ? value : undefined),
+          )
+
+          expect(defaultMatchingInformationValues(bodyWithUndefinedValues, assessment)).toEqual(
+            expect.objectContaining({ isSuitedForSexOffenders: 'essential' }),
+          )
+        },
+      )
+    })
```

Comment:
This generates 12 tests (4 `sexualOffencesFields` x 3 `truthyValues`): this might be overkill, and we aren't testing what happens if the values are arrays not containing either value, but that shouldn't happen as far as I understand it